### PR TITLE
Added calums user ingredient validator

### PR DESCRIPTION
**Pull request by jdselby24**

### CODE REVIEW

File: src/UserIngredientValidator.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+*/
+abstract class UserIngredientValidator {
+	private $validatedIngredients = [];
+	
+	/**
+	* Function to validate ingredients input by user and return a 'clean' array of ingredients with whitespace removed
+	*
+	* @param array $explodedUserIngredients an array of the user ingredients submitted from index.php
+	*
+	* @returns array of validated user-inputted ingredients
```
Comment (mgallagher56) [inline src/UserIngredientValidator.php]:
```suggestion
	* @return array of validated user-inputted ingredients
```

File: src/UserIngredientValidator.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace GRUB;
+/*
+* Abstract class with a static function used for validation of user-inputted ingredients
+*/
```
Comment (mgallagher56) [inline src/UserIngredientValidator.php]:
```/**
 * Class UserIngredientValidator class with a static function used for validation of user-inputted ingredients
 * 
 * @packages GRUB
 */```

Comment (mgallagher56) [inline src/UserIngredientValidator.php]:
```suggestion
/**
 * Class UserIngredientValidator class with a static function used for validation of user-inputted ingredients
 * 
 * @packages GRUB
 */
```

File: src/UserIngredientValidator.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+abstract class UserIngredientValidator {
+	private $validatedIngredients = [];
```
Comment (mgallagher56) [inline src/UserIngredientValidator.php]: Doc block

File: src/UserIngredientValidator.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+abstract class UserIngredientValidator {
+    /**
+	* Function to validate ingredients input by user and return a 'clean' array of ingredients with whitespace removed
+	*
+	* @param array $explodedUserIngredients an array of the user ingredients submitted from index.php
+	*
+	* @returns array of validated user-inputted ingredients
+	*/
+	public static function validateUserIngredients(array $explodedUserIngredients) :array {
+        private $validatedIngredients = [];
```
Comment (mgallagher56) [inline src/UserIngredientValidator.php]:
```suggestion
        $validatedIngredients = [];
```

File: src/UserIngredientValidator.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+		if ($explodedUserIngredients != []){
+			foreach ($explodedUserIngredients as $ingredient){
+				//removes anything that isn't a letter and removes all whitespace
+				//the letter i at the end is used to remove case sensitivity from the regex
+				$validatedIngredient = preg_replace( '/[^a-z]/i', '', $ingredient);
+				array_push($validatedIngredients, $validatedIngredient);
+			}
+			return $validatedIngredients;
+		} else {
+			return "Invalid data. Please try again.";
```
Comment (mgallagher56) [inline src/UserIngredientValidator.php]: Should this be echo + html to output an error message?

Comment (cwoodwarduk) [reply src/UserIngredientValidator.php]: Done.

**Pull request merged**